# Anatomy of a Flop — Preprocessing

## Scopul
Curățăm datele și le pregătim pentru modelare.

### Pași:
1. Eliminarea valorilor lipsă
2. Feature engineering
3. Salvarea datelor curate

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv('../data/movies_raw.csv')
print(f" Date încărcate: {df.shape}")

 Date încărcate: (1000, 18)


In [2]:
# Eliminăm filmele fără date financiare
df_clean = df[(df['budget'] > 100000) & (df['revenue'] > 100000)].copy()
print(f" filme cu date complete: {len(df_clean)}")

# Disappointment Index
df_clean['expected_revenue'] = df_clean['budget'] * 2.5
df_clean['disappointment_index'] = df_clean['revenue'] / df_clean['expected_revenue']
df_clean['roi'] = (df_clean['revenue'] - df_clean['budget']) / df_clean['budget']
df_clean['release_year'] = pd.to_datetime(df_clean['release_date']).dt.year
df_clean['release_month'] = pd.to_datetime(df_clean['release_date']).dt.month

print(df_clean[['budget','revenue','disappointment_index','roi']].describe())

 filme cu date complete: 866
             budget       revenue  disappointment_index         roi
count  8.660000e+02  8.660000e+02            866.000000  866.000000
mean   8.310540e+07  3.606459e+08              3.005975    6.514936
std    7.746028e+07  3.805884e+08              5.531719   13.829298
min    3.250000e+05  1.243670e+05              0.000908   -0.997730
25%    2.000000e+07  8.902500e+07              0.984993    1.462482
50%    6.000000e+07  2.449706e+08              1.667067    3.167668
75%    1.300000e+08  5.047626e+08              2.809941    6.024851
max    4.899000e+08  2.923706e+09             86.474581  215.186452


### Observații după curățare

- **866 filme** au date financiare complete din 1000
- **DI mediu: 3.0** — în medie filmele câștigă de 3x față de așteptări
- **ROI mediu: 6.5** — returnul investiției e de 650% în medie
- **Std DI foarte mare (5.5)** — variații enorme între filme
- **Min ROI: -0.99** → există filme care au pierdut aproape tot bugetul

In [3]:
# Salvăm datele curate
df_clean.to_csv('../data/movies_clean.csv', index=False)
print(f" Salvat! {len(df_clean)} filme în movies_clean.csv")

 Salvat! 866 filme în movies_clean.csv



### Ce am făcut concret:

**1. Eliminarea valorilor lipsă**
- Din 1000 filme originale, 134 aveau budget = 0 sau revenue = 0
- Acestea nu reprezintă filme gratuite ci **date lipsă din TMDB**
- Am păstrat doar cele 866 filme cu date financiare complete

**2. Feature Engineering — variabile noi create:**
- **`expected_revenue`** = budget × 2.5
  → cât ar trebui să câștige un film ca să fie profitabil
- **`disappointment_index`** = revenue real / expected_revenue
  → variabila Y pe care o prezice modelul nostru
- **`roi`** = (revenue - budget) / budget
  → returnul investiției în procente
- **`release_year`** și **`release_month`**
  → extrase din data lansării pentru a capta sezonalitatea

**3. Salvarea datelor**
- `movies_raw.csv` → datele originale din TMDB
- `movies_clean.csv` → datele curate, gata pentru modelare

### Concluzie:
> Am transformat 1000 de înregistrări brute în 866 observații
> curate cu variabile relevante pentru modelul nostru.